src/uu/df/df.rs

```diff
@@ -0,0 +1,878 @@
+        loop {
+            let volume_name = LPWSTR2String(&volume_name_buf);
+            if !volume_name.starts_with("\\\\?\\") || !volume_name.ends_with("\\") {
+                show_warning!("A bad path was skipped: {}", volume_name);
+                continue;
+            }
+            if let Some(m) = MountInfo::new(volume_name) {
+                mounts.push(m);
+            }
+            if 0 == unsafe {
+                FindNextVolumeW(
+                    find_handle,
+                    volume_name_buf.as_mut_ptr(),
+                    volume_name_buf.len() as winapi::DWORD,
+                )
+            } {
+                let err = unsafe { GetLastError() };
+                if err != winapi::ERROR_NO_MORE_FILES {
+                    crash!(EXIT_ERR, "FindNextVolumeW failed: {}", err);
+                }
+                break;
+            }
+        }
+        unsafe {
+            FindVolumeClose(find_handle);
+        }
+        return mounts;
+    }
+}
+
+fn filter_mount_list(vmi: Vec<MountInfo>, opt: &Options) -> Vec<MountInfo> {
+    vmi.into_iter()
+        .filter_map(|mi| {
+            if (mi.remote && opt.show_local_fs)
+                || (mi.dummy && !opt.show_all_fs && !opt.show_listed_fs)
+                || !opt.fs_selector.should_select(&mi.fs_type)
+            {
+                None
+            } else {
+                Some((mi.dev_id.clone(), mi))
+            }
+        })
+        .fold(
+            HashMap::<String, Cell<MountInfo>>::new(),
+            |mut acc, (id, mi)| {
+                if acc.contains_key(&id) {
+                    let seen = acc.get(&id).unwrap().replace(mi.clone());
+                    let target_nearer_root = seen.mount_dir.len() > mi.mount_dir.len();
+                    // With bind mounts, prefer items nearer the root of the source
+                    let source_below_root = !seen.mount_root.is_empty()
+                        && !mi.mount_root.is_empty()
+                        && seen.mount_root.len() < mi.mount_root.len();
+                    // let "real" devices with '/' in the name win.
+                    if (!mi.dev_name.starts_with('/') || seen.dev_name.starts_with('/'))
+                // let points towards the root of the device win.
+                && (!target_nearer_root || source_below_root)
+                // let an entry overmounted on a new device win...
+                && (seen.dev_name == mi.dev_name
+                    /* ... but only when matching an existing mnt point,
+                       to avoid problematic replacement when given
+                       inaccurate mount lists, seen with some chroot
+                       environments for example.  */
+                    || seen.mount_dir != mi.mount_dir)
+                    {
+                        acc.get(&id).unwrap().replace(seen);
+                    }
+                } else {
+                    acc.insert(id, Cell::new(mi));
+                }
```

Comment:
```shell
warning: usage of `contains_key` followed by `insert` on a `HashMap`
   --> src/uu/df/df.rs:559:17
    |
559 | /                 if acc.contains_key(&id) {
560 | |                     let seen = acc.get(&id).unwrap().replace(mi.clone());
561 | |                     let target_nearer_root = seen.mount_dir.len() > mi.mount_dir.len();
562 | |                     // With bind mounts, prefer items nearer the root of the source
...   |
581 | |                     acc.insert(id, Cell::new(mi));
582 | |                 }
    | |_________________^ consider using `acc.entry(id)`
    |
    = note: `#[warn(clippy::map_entry)]` on by default
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#map_entry
```

Reply:
I am ignoring this error.
I don't think clippy is right in that case.

Reply:
I'm fine with that ... just add an `#[allow(clippy::map_entry)]` on that line/block.

We can, in the future, occasionally review all the `allow`s for possible refactoring, if needed.

Reply:
Clippy doesn't support ignore on a if, I had to do it as the function level

Reply:
Make it a block by surrounding it with brackets, then you can apply the attribute to the block. That will narrow the code area window.